scriptmodules/ports/hexen2.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+
+# This file is part of The RetroPie Project
+#
+# The RetroPie Project is the legal property of its developers, whose names are
+# too numerous to list here. Please refer to the COPYRIGHT.md file distributed with this source.
+#
+# See the LICENSE.md file at the top-level directory of this distribution and
+# at https://raw.githubusercontent.com/RetroPie/RetroPie-Setup/master/LICENSE.md
+#
+
+rp_module_id="hexen2"
+rp_module_desc="Hexen II - Hammer of Thyrion source port"
+rp_module_licence="GLP2"
+rp_module_help="For registered version, please add your full version PAK files to $romdir/ports/hexen2/data1/ to play. These files for the registered version are required: pak0.pak, pak1.pak and strings.txt. The registered pak files must be patched to 1.11 for Hammer of Thyrion."
+rp_module_section="opt"
+rp_module_flags=""
+
+
+# Get depends required by hexen2
+function depends_hexen2() {
+    getDepends cmake libsdl1.2debian libsdl-image1.2 libsdl-mixer1.2 libsdl-mixer1.2-dev libsdl-net1.2 libsdl-net1.2-dev timidity
+}
+
+# Clone from uhexen2 source
+function sources_hexen2() {
+    gitPullOrClone "$md_build" https://github.com/svn2github/uhexen2.git
+}
+
+# Build hexen2
+function build_hexen2() {
+    cd "$md_build/engine/hexen2"
+    ./build_all.sh
+    md_ret_require="$md_build/engine/hexen2/hexen2"
+}
+
+function install_hexen2() {
+    md_ret_files=(        
+        'engine/hexen2/hexen2'
+    )
+}
+
+# Download hexen2 shareware version
+function game_data_hexen2() {
+	if [[ ! -f "$romdir/ports/hexen2/pak0.pak" ]]; then
+        downloadAndExtract "https://netix.dl.sourceforge.net/project/uhexen2/Hexen2Demo-Nov.1997/hexen2demo_nov1997-linux-i586.tgz" "$romdir/ports/hexen2"
+		mv /home/pi/RetroPie/roms/ports/hexen2/hexen2demo_nov1997/data1 /home/pi/RetroPie/roms/ports/hexen2
+		rm -r /home/pi/RetroPie/roms/ports/hexen2/hexen2demo_nov1997
+    fi
+    # always chown as moveConfigDir in the configure_ script would move the root owned demo files
+    chown -R $user:$user "$romdir/ports/hexen2"
+}
+
+# Configure hexen2 RetroPie
+function configure_hexen2() {
+    mkRomDir "ports"
+    mkRomDir "ports/hexen2"
+
+    mkUserDir "$home/.hexen2"
```

Comment:
can't this be linked to `$confdir/ports/hexen2` instead ?

Reply:
What does $confdir relate to?
Is it this dir /home/pi/.config?

Reply:
apologies, the correct variable name is `$configdir`.
This variable points to `/opt/retropie/configs` so you can store game config files in `$configdir/ports/hexen2`.  Normally, you want to link `$HOME/.hexen2` to `$configdir/ports/hexen2`. In this case your function should be:

    moveConfigDir "$home/.hexen2" "$configdir/ports/hexen2"

Reply:
Yes that sounds correct but when comparing to Hexen 2 Hammer of Thyrion source port i think it will be better if the "$home/.hexen2" is linked to "$romdir/ports/hexen2", since hexen2 binary will be looking for the game files in this dir: "home/pi/.hexen2/data1/" or "$home/.hexen2/data1/".

Maybe you could first link "$home/.hexen2" to $configdir/ports/hexen2 and then link $configdir/ports/hexen2 to "$romdir/ports/hexen2". Would this be an option?.